packages/web/src/features/translation/verse-utils.ts

```diff
@@ -132,38 +133,56 @@ export function bookName(bookId: number, t: TFunction) {
   return t(bookKeys[bookId - 1], { ns: 'bible' });
 }
 
+/**
+ * Convert a reference format string for a language to a reference matching regex for that language.
+ * @param formatString The reference format string. See i18n.md for docs.
+ * @returns A regex for matching references.
+ */
+export function convertFormatToRegex(formatString: string): RegExp {
+  // TODO: improve this to increase language support.
```

Comment:
I left this for the future, if we find languages that need additional things beyond what this conversion supports.